### PR TITLE
MBS-7908: ignore non-library dependencies in impact analysis

### PR DESCRIPTION
**Pull request by dsvoronin**

### CODE REVIEW

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt [hunk 1]
```diff
@@ -58,6 +59,10 @@ class ImplementationConfiguration(module: InternalModule) : SimpleConfiguration(
                     .dependencies
                     .withType(DefaultProjectDependency::class.java)
             }
+            .filter {
+                val category = it.attributes.getAttribute(Category.CATEGORY_ATTRIBUTE)?.name
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt:63]: It is not clear line. Can you explain what is a `Category` and `CATEGORY_ATTRIBUTE`?

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt [hunk 2]
```diff
@@ -58,6 +59,10 @@ class ImplementationConfiguration(module: InternalModule) : SimpleConfiguration(
...
+                category == null || category == Category.LIBRARY
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt:64]: What deps are with category null?

Comment (dsvoronin) [reply subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt:64]: most of them is null (i just used debugger to check, don't know much)

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/ImplementationConfiguration.kt:63]: https://docs.gradle.org/current/javadoc/org/gradle/api/attributes/Category.html

**Pull request merged**